spring-cloud-aws-autoconfigure/src/test/java/io/awspring/cloud/autoconfigure/s3/S3AutoConfigurationTests.java

```diff
@@ -69,6 +74,26 @@ void createsS3ClientBean() {
 		});
 	}
 
+	@Test
+	void createsS3TransferManagerBeanWhenInClassPath() {
+		this.contextRunner.run(context -> assertThat(context).hasSingleBean(S3TransferManager.class));
+	}
+
+	@Test
+	void usesExistingS3TransferManagerBeanWhenExists() {
+		S3TransferManager customDefinedS3TransferManager = Mockito.mock(S3TransferManager.class);
+		this.contextRunner.withBean("s3transferManager", S3TransferManager.class, () -> customDefinedS3TransferManager)
```

Comment:
`withBean` - this is much nicer way to register custom beans than what we've been doing before!